### PR TITLE
Fixed error in SqlQueryValidator with multiple conditions in ON clause

### LINKED ISSUE #25684: BO - SQL Manager - SQL query return exception
#### Describe the bug

SQL Manager throws Fatal Error when trying to run complex query

#### Expected behavior

No exception should be displayed

#### Steps to Reproduce

1 - Go to BO => discount
2 - Click on "export to sql manager" button
3 - In the SQL manager page, just click on "save" button
4- See error

**Screenshots**

https://drive.google.com/file/d/152rsnZDnxUPT38OesuWXL5pCuYDE9-1f/view?usp=sharing

#### Additional information

* PrestaShop version: 1.7.7.x // 1.7.8.x
* PHP version: 7.3 // 7.4

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Fixed error in SqlQueryValidator with multiple conditions in ON clause
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25684
| How to test?      | Cf #25684

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25760)

### CODE REVIEW

File: classes/RequestSql.php [hunk 1]
```diff
@@ -239,8 +239,20 @@ public function cutJoin($attrs, $from)
             if (in_array($attr['expr_type'], ['operator', 'const'])) {
                 continue;
             }
-            if ($attribut = $this->cutAttribute($attr['base_expr'], $from)) {
-                $tab[] = $attribut;
+
+            if (!empty($attr['sub_tree'])) {
+                foreach ($attr['sub_tree'] as $treeItem) {
+                    if ($treeItem['expr_type'] !== 'colref') {
```
Comment (sowbiba) [inline classes/RequestSql.php:245]: If I understand well, colref is a reference for DB column. Can you make a test with the request in the issue ( sthg like a.x = b.x AND 1) ?

**Review by sowbiba — COMMENTED** [review]
I'm curious. The issue says it's a regression from 1775. Does it means the logic you add in this PR was removed in 1775 ?

File: classes/RequestSql.php (hunk 1 above)
Comment (Progi1984) [reply classes/RequestSql.php:245]: I have added a case ;)

**Review by sowbiba — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

File: tests/Integration/Adapter/SqlManager/SqlQueryValidatorTest.php [hunk 2]
```diff
@@ -59,6 +59,22 @@ public function dataProviderValidate(): iterable
...
             [],
         ];
+        yield [
+            'SELECT SQL_CALC_FOUND_ROWS b.*, a.* '
+            . 'FROM `ps_cart_rule` a '
+            . 'LEFT JOIN `ps_cart_rule_lang` b ON (b.`id_cart_rule` = a.`id_cart_rule` AND b.`id_lang` = 1) '
+            . 'WHERE 1 '
+            . 'ORDER BY a.id_cart_rule DESC',
+            [],
+        ];
```
Comment (PierreRambaud) [inline tests/Integration/Adapter/SqlManager/SqlQueryValidatorTest.php:69]:
```suggestion
        yield [
            'SELECT SQL_CALC_FOUND_ROWS b.*, a.* '
            . 'FROM `ps_cart_rule` a '
            . 'LEFT JOIN `ps_cart_rule_lang` b ON (b.`id_cart_rule` = a.`id_cart_rule` AND b.`id_lang` = 1) '
            . 'WHERE 1 '
            . 'ORDER BY a.id_cart_rule DESC',
            [],
        ];
        yield [
            'SELECT SQL_CALC_FOUND_ROWS b.*, a.* '
            . 'FROM `ps_cart_rule` a '
            . 'LEFT JOIN `ps_cart_rule_lang` b ON (b.`id_cart_rule` = a.`id_cart_rule` AND b.`id_lang` = a.`id_lang`) '
            . 'WHERE 1 '
            . 'ORDER BY a.id_cart_rule DESC',
            [],
        ];
```

Comment (PierreRambaud) [reply tests/Integration/Adapter/SqlManager/SqlQueryValidatorTest.php:69]: I experiment differents behaviors when we have something different than 1. To be sure everything is ok

Comment (Progi1984) [reply tests/Integration/Adapter/SqlManager/SqlQueryValidatorTest.php:69]: Test added.

**Review by PierreRambaud — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

It is ok :heavy_check_mark: 

https://watch.screencastify.com/v/E9lOGfDXPOOrMeUDc8Qp

I checked:

- Export of legacy page (cart rule page, catalog price rule page, zones page, Attributes page)
- Export of a migrated page (orders page, webservice page, logs page, SQL page)
- Create SQL of delete
- Create SQL with incorrect data

Thanks!

**Comment by Progi1984** [comment]
Thanks @khouloudbelguith

**Pull request merged**